### PR TITLE
Migrate v3.6.0 release notes to codebase

### LINKED ISSUE #15434: Migrate release notes for version 3.6 to the repository
It was decided by the [TSC](https://github.com/mantidproject/documents/blob/master/Project-Management/TechnicalSteeringCommittee/meetings/2016/TSC-meeting-2016-02-23.md) that as of version `3.7` the release notes should be part of the offline documentation. This will allow edits to the release notes to stay in sync with the code.

It was decided that the notes for `3.6` should be ported in order to provide a set of illustrative examples for developers moving forward.

### PR DESCRIPTION
**Pull request by martyngigg**
The release notes for v3.6.0 have been migrated to reStructuredText and integrated with the existing docs. As a result they will now be shipped with the code. The main purpose here is to have an illustrative set of examples for including release notes in future versions.

**To test:**
- Build the HTML documentation and view it locally in a web browser
- Check the navigation through from the main index page to each of the sub topics
- Check the formatting of each page is reasonable and all images have been included
- Check you agree with how the files have been named etc within the source code and resulting urls

The wiki version of the notes is [here](http://www.mantidproject.org/Release_Notes_3.6) for comparison.

Fixes #15434

Does not need to be in the release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the coding standards? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [x] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by peterfpeterson** [comment]
The new documentation looks good. Assuming that the builds go through I think we are ready to :shipit:. The main hiccup now is moving the release notes for 3.7 over.

File: .gitignore [hunk 1]
```diff
@@ -77,8 +77,6 @@ local.properties
 *.log
 
 # Build results
-[Dd]ebug/
-[Rr]elease/
```
Comment (rosswhitfield) [inline .gitignore]: Why? Couldn't this just have been changed to `/[Rr]elease/` and `/[Dd]ebug/`?

Comment (martyngigg) [inline .gitignore]: It could. I thought those were hangovers from olden days so I just got rid. Shall I add them back?

Comment (rosswhitfield) [inline .gitignore]: Well I use them. But I guess I could switch to one of the other build directories that we ignore.

Comment (martyngigg) [inline .gitignore]: No, let's just add `/[Rr]elease/` and `/[Dd]ebug/` to `.gitignore`. Sorry for lack of consideration.

Comment (martyngigg) [inline .gitignore]: Done in https://github.com/mantidproject/mantid/commit/87326e78f950ec70229afcc4dca74db3669771d8

Comment (rosswhitfield) [inline .gitignore]: :+1: Thanks.

**Pull request merged**